### PR TITLE
Various fixes for the DPsim Interface

### PR DESCRIPTION
**Pull request by n-eiling**
The DPsim interface required a few fixes in VILLAS.
Some are not strictly necessary, but I think the extra checks make the code more robust.

### CODE REVIEW

File: lib/hooks/dp.cpp [hunk 1]
```diff
@@ -233,7 +233,7 @@ class DPHook : public Hook {
       signals->erase(signals->begin() + signal_index);
 
       for (int i = 0; i < fharmonics_len; i++) {
-        new_sig_name = strf("%s_harm%d", orig_sig->name, i);
+        new_sig_name = strf("%s_harm%d", orig_sig->name.c_str(), i);
```
Comment (stv0g) [inline lib/hooks/dp.cpp:236]: Would be great to switch to `fmt::format()` here

File: lib/nodes/fpga.cpp [hunk 2]
```diff
@@ -341,6 +341,11 @@ int FpgaNode::slowWrite(Sample *smps[], unsigned cnt) {
 }
 
 std::vector<int> FpgaNode::getPollFDs() {
+  if (!lowLatencyMode && card && !card->polling) {
+    return card->vfioDevice->getEventfdList();
+  } else {
+    return {};
+  }
   return card->vfioDevice->getEventfdList();
```
Comment (stv0g) [inline lib/nodes/fpga.cpp]: Seems like this is dead code? Can we remove it? Or just move the else branch out?

**Review by stv0g — CHANGES_REQUESTED** [review]

**Review by stv0g — CHANGES_REQUESTED** [review]

**Comment by stv0g** [comment]
Thanks :)

**Review by stv0g — APPROVED** [review]

**Comment by n-eiling** [comment]
Thanks for the fast review. I was a bit lazy on the fmt::format thing, but porting to C++ was easier than I thought.

**Pull request merged**